distributed/comm/ucx.py

```diff
@@ -39,6 +43,50 @@ def init_once():
     import ucp as _ucp
 
     ucp = _ucp
+
+    # configuration of UCX can happen in two ways:
+    # 1) high level on/off flags which correspond to UCX configuration
+    # 2) explicity defined UCX configuration flags
+
+    # if any of the high level flags are set, as long as they are not Null/None,
+    # we assume we should configure basic TLS settings for UCX
+
+    if any(
+        [
+            dask.config.get("ucx.nvlink"),
+            dask.config.get("ucx.infiniband"),
+            dask.config.get("ucx.tcp-over-ucx"),
+            dask.config.get("ucx.net-devices"),
+        ]
+    ):
+        tls = "tcp,sockcm,cuda_copy"
+        tls_priority = "sockcm"
+
+        if dask.config.get("ucx.infiniband"):
+            tls = "rc," + tls
+        if dask.config.get("ucx.nvlink"):
+            tls = tls + ",cuda_ipc"
+
+        options = {"TLS": tls, "SOCKADDR_TLS_PRIORITY": tls_priority}
+
+        net_devices = dask.config.get("ucx.net-devices")
+        if net_devices is not None and net_devices != "":
+            options["NET_DEVICES"] = net_devices
+
+        dask.config.set({"ucx": options})
+
+    # support other clients (dask-cuda) passing ucx configurations directly
+    # when using explicty UCX configurations assume advanced users
+    elif all(
+        [
+            dask.config.get("ucx.TLS", False),
+            dask.config.get("ucx.SOCKADDR_TLS_PRIORITY", False),
```

Comment:
Should these default values be in the yaml file instead?

Reply:
Admittedly this may be a conservative opinion, I don't think so.  There are loads of UCX config options and this could easily explode.  On there hand, maybe a minimal set _might_ be ok.   @pentschev thoughts ?

env vars which have been important for us lately would include:
- TLS
- SOCKADDR_TLS_PRIORITY
- MEMTYPE_CACHE
- CUDA_IPC_CACHE
- NET_DEVICES

Reply:
Hmm, the way I was handling defaults is not to have defaults, i.e., the user is forced to select at least one option. The reason for doing that is that users are then aware of what they're selecting until we are confident options work more broadly. IMO, this is the best way to do it now, as users will know they _need_ to specify what they want to use rather than assume UCX will enable everything by default -- maybe one day that will be the case, but that day is still not today, unfortunately.

Reply:
To be more specific, if the user doesn't choose at least TCP over UCX, dask-cuda will error out today.